test/e2e/app/snapshots.go

```diff
@@ -105,6 +108,23 @@ func (s *SnapshotStore) Create(state *State) (abci.Snapshot, error) {
 	return snapshot, nil
 }
 
+// Prune removes old snapshots ensuring only the most recent n snapshots remain
+func (s *SnapshotStore) Prune(n int) error {
+	s.Lock()
+	defer s.Unlock()
+	// snapshots are appended to the metadata struct, hence pruning removes from
+	// the front of the array
+	i := 0
+	for ; i < len(s.metadata)-n; i++ {
+		h := s.metadata[i].Height
+		if err := os.Remove(filepath.Join(s.dir, fmt.Sprintf("%v.json", h))); err != nil {
+			return err
+		}
+	}
+	s.metadata = s.metadata[i:]
```

Comment:
Because the underlying array will not be collected until all the slices of it have pointed away, it's usually best to shift and reslice or allocate a new array when discarding a prefix of a slice:

```suggestion
	// reslice
	n := copy(s.metadata, s.metadata[i:])
	s.metadata = s.metadata[:n]
	
	// or, reallocate
	pruned := make([]metadata, len(s.metadata[i:]))
	copy(pruned, s.metadata[i:])
	s.metadata = pruned
```

This is not likely to be a major issue in this context, since we probably have much bigger memory footprints around, but the pattern is easy enough to fix.